### PR TITLE
Add test cases for conditions

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
**Type of PR:**
-  Other

**Required reviews:**
- 1

**What this does:**
> High-level idea of the changes introduced in this PR.
> List relevant API changes (if any), as well as related PRs and issues.

**Issues fixed/closed:**
> - Fixes #...

**Why it's needed:**
> Explain how this PR fits in the greater context of the NuCypher Network.
> E.g., if this PR address a `nucypher/productdev` issue, let reviewers know!

**Notes for reviewers:**
- Based on top of #379 
- Ideas for testing cases welcome

### CODE REVIEW

File: packages/taco/test/conditions/base/rpc.test.ts [hunk 1]
```diff
@@ -94,5 +94,41 @@ describe('validation', () => {
         },
       });
     });
+
+    it('rejects empty parameters', () => {
```
Comment (derekpierre) [inline packages/taco/test/conditions/base/rpc.test.ts:98]: This is fine for now, but something to note/update as we expand to allow other rpc methods.

File: packages/taco/test/conditions/compound-condition.test.ts [hunk 2]
```diff
@@ -216,4 +216,36 @@ describe('validation', () => {
...
+
+  it('rejects non-array operands for "and" and "or" operators', () => {
+    const badObj = {
+      operator: 'and',
+      operands: testContractConditionObj,
+    };
+    expect(() => new CompoundCondition(badObj)).toThrow();
+  });
+
+  it('rejects array operands for "not" operator', () => {
```
Comment (derekpierre) [inline packages/taco/test/conditions/compound-condition.test.ts:244]: To clarify, the following still holds, correct?
- `not`, `and`, and `or` all accept array operands, just that `not` requires the array to be size 1, and `and`/`or` expect array size to be > 1.
- They all should fail if "operands" is not an array.

File: packages/taco/test/conditions/context.test.ts [hunk 3]
```diff
@@ -262,15 +262,21 @@ describe('param or context param schema', () => {
   });
 
   it('accepts a hex string', () => {
-    expect(paramOrContextParamSchema.safeParse("deadbeef").success).toBe(true);
+    expect(paramOrContextParamSchema.safeParse('deadbeef').success).toBe(true);
```
Comment (derekpierre) [inline packages/taco/test/conditions/context.test.ts:265]: Careful here - any hex string as a representation of bytes should be prefixed by `0x`. Strings that are just strings don't need the `0x` prefix.

File: packages/taco/test/conditions/compound-condition.test.ts (hunk 2 above)
Comment (piotr-roslaniec) [reply packages/taco/test/conditions/compound-condition.test.ts:244]: I've added some tests to cover these cases

Comment (derekpierre) [reply packages/taco/test/conditions/compound-condition.test.ts]: 👍

File: packages/taco/test/conditions/compound-condition.test.ts [hunk 4]
```diff
@@ -205,4 +216,67 @@ describe('validation', () => {
...
+
+  it('rejects invalid operator', () => {
+    const badObj = {
+      operator: 'invalid-operator',
+      operands: [testContractConditionObj, testTimeConditionObj],
+    };
+    expect(() => new CompoundCondition(badObj)).toThrow();
+  });
+
+  it.each(['or', 'and'])(
```
Comment (derekpierre) [inline packages/taco/test/conditions/compound-condition.test.ts:228]: Should `not` also be in here - it should reject empty operands as well?
```suggestion
  it.each(['or', 'and', 'not'])(
```

File: packages/taco/test/conditions/compound-condition.test.ts [hunk 5]
```diff
@@ -205,4 +216,67 @@ describe('validation', () => {
...
+    (operator) => {
+      const badObj = {
+        operator,
+        operands: [],
+      };
+      expect(() => new CompoundCondition(badObj)).toThrow();
+    },
+  );
+
+  it.each(['or', 'and'])(
```
Comment (derekpierre) [inline packages/taco/test/conditions/compound-condition.test.ts:239]: Should `not` be in here as well - it should reject non-array operands?
```suggestion
  it.each(['or', 'and', 'not'])(
```

**Review by manumonti — APPROVED** [review]
LGTM!

**Pull request merged**